http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,246 @@
+                            .compose(req -> req.send()
+                                    .compose(httpClientResponse -> {
+                                        requestCheckpoint.flag();
+                                        assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                                        assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                                        return httpClientResponse.body();
+                                    }))
+                            .onSuccess(body -> {
+                                assertThat("Body response", body.toString().contains(GREETING));
+                                requestCheckpoint.flag();
+                            }).onFailure(throwable -> vertxTestContext.failNow(throwable.getMessage()));
+                });
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("HttpClient Vertx just with HTTP/2 protocol option set and default port")
+    void http2ProtocolTest(Vertx vertx, VertxTestContext vertxTestContext) {
+        HttpClientOptions httpClientOptions = new HttpClientOptions()
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(httpClientOptions).build();
+        httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(request -> request.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(HttpVersion.HTTP_2, httpClientResponse.version());
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> vertxTestContext.verify(() -> {
+                    assertThat("Body response", body.toString().contains(GREETING));
+                    vertxTestContext.completeNow();
+                })).onFailure(Throwable::printStackTrace);
+
+    }
+
+    @Test
+    @DisplayName("Test max header size exceeded error message")
+    @Order(5)
+    void testMaxHeaderSizeExceeded(Vertx vertx, VertxTestContext testContext) {
+        HttpClientOptions options = new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustOptions(new JksOptions().setPath("keystore.jks").setPassword("password"))
+                .setProtocolVersion(HttpVersion.HTTP_2);
+        httpClient = vertx.httpClientBuilder().with(options).build();
+        // Define the expected maximum header size limit
+        // Note: Consider the configured limit, which is 2048 in this case because we are testing quarkus.http.limits.max-header-list-size=2048,
+        // although it may be considerably lower in practice
```

Comment:
I like the note, it's almost correct. I think you confused number of headers with overall headers size. The name of the exception is confusing, but I believe that Netty is referring to the headers size in byte, so when you say 2048 it means total size of headers you generated